homeassistant/components/ista_ecotrend/sensor.py

```diff
@@ -184,3 +196,85 @@ def native_value(self) -> StateType:
             consumption_type=self.entity_description.consumption_type,
             value_type=self.entity_description.value_type,
         )
+
+    async def async_added_to_hass(self) -> None:
+        """When added to hass."""
+        # perform initial statistics import when sensor is added, otherwise it would take
+        # 1 day when _handle_coordinator_update is triggered for the first time.
+        await self.update_statistics()
+        await super().async_added_to_hass()
+
+    def _handle_coordinator_update(self) -> None:
+        """Handle coordinator update."""
+        asyncio.run_coroutine_threadsafe(self.update_statistics(), self.hass.loop)
+
+    async def update_statistics(self) -> None:
+        """Import ista EcoTrend historical statistics."""
+        if TYPE_CHECKING:
+            assert self.coordinator.config_entry
```

Comment:
why don't we overwrite the type in the coordinator?

Reply:
Indeed a very good question 😬 
Also improved typing for device_entry while i was at it.